internal/query/user.go

```diff
@@ -655,7 +655,11 @@ func (q *Queries) searchUsers(ctx context.Context, queries *UserSearchQueries, f
 		UserInstanceIDCol.identifier(): authz.GetInstance(ctx).InstanceID(),
 	})
 	if permissionCheckV2 {
-		query = wherePermittedOrgs(ctx, query, filterOrgIds, UserResourceOwnerCol.identifier(), domain.PermissionUserRead)
+		if allowReturnCurrentUser {
```

Comment:
We don't need to make this conditional. The current user should always be returned by searchUsers. That's the bug we are solving here.